[java] set isDisabled to true by default (make Bindings opt-in)

titusfortner opened the pull request:
I think we want to make Sauce Bindings opt-in instead of opt-out.

This has the advantage of not changing default behavior when registering extension or adding rule.
This would require the user to update their CI tool to run with an environment variable or system property.
This would require the user to add `static { System.setProperty("sauce.disabled", "false") }` in order to run tests on Sauce Labs.
If we have this logic in CI plugins (e.g. Jenkins), it would just work on CI. If we create IDE plugins, we just add the toggle to the Run configs or context clicks.

Now that we are making this switch, do we want to make this `sauce.enabled` instead? Probably.

titusfortner committed:
[java] set isDisabled to true by default (make Bindings opt-in)

diemol reviewed (commented):
Yeah, I would change the name to enabled.

titusfortner committed:
[java] update the property for enabling Sauce Bindings

titusfortner committed:
[java] revamp tests for bindings to be opt-in

titusfortner commented:
Ok, so part of making things opt-in...
I switched from maven executions to profiles and updated the Github actions workflows accordingly.

I added the convenience method for enabling the extension rather than requiring a system property, but added the details in the code comments.

Really when it comes down to it, I want to make sure that these examples are what we want users to do:
https://github.com/saucelabs/sauce_bindings/tree/513cd55a16b33c73e104b9011bdef05075518371/java/junit5/src/test/java/com/saucelabs/saucebindings/junit5/examples

diemol reviewed (approved):
I'd say this is good enough.
